app/src/main/java/com/zulip/android/activities/DevAuthActivity.java

```diff
@@ -0,0 +1,147 @@
+        connectionProgressDialog.dismiss();
+        Intent i = new Intent(this, ZulipActivity.class);
+        startActivity(i);
+        finish();
+    }
+    public class AuthEmailAdapter extends RecyclerView.Adapter<DevAuthActivity.AuthEmailViewHolder> {
+
+        private static final int VIEW_TYPE_HEADER = 0;
+        private static final int VIEW_TYPE_ADMIN = 1;
+        private static final int VIEW_TYPE_USER = 2;
+        List<String> emails;
+        int direct_admin_size;
+
+        public AuthEmailAdapter(List<String> emails, int direct_admin_size) {
+            this.direct_admin_size = direct_admin_size;
+            this.emails = emails;
+        }
+
+        @Override
+        public AuthEmailViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+            View v = null;
+            if (viewType == VIEW_TYPE_HEADER)
+                v = getLayoutInflater().inflate(R.layout.dev_auth_header, parent, false);
+            else
+                v = getLayoutInflater().inflate(R.layout.dev_auth_row, parent, false);
+
+            AuthEmailViewHolder authEmailViewHolder = new AuthEmailViewHolder(v);
+            return authEmailViewHolder;
+        }
+
+        @Override
+        public void onBindViewHolder(AuthEmailViewHolder holder, int position) {
+            if (position == 0) {
+                if (direct_admin_size == 0)
+                    ((TextView) holder.view).setText(R.string.admin_none); //Show none if no admins
+                else ((TextView) holder.view).setText(R.string.admin);
+            } else if (position == direct_admin_size + 1)
+                ((TextView) holder.view).setText(R.string.normal_user);
+            else if (position > direct_admin_size + 1) {
+                ((Button) holder.view).setText(emails.get(position - 2));
+                ((Button) holder.view).setBackgroundColor(ContextCompat.getColor(DevAuthActivity.this, R.color.dev_auth_user));
+            } else {
+                ((Button) holder.view).setText(emails.get(position - 1));
+                ((Button) holder.view).setBackgroundColor(ContextCompat.getColor(DevAuthActivity.this, R.color.dev_auth_admin));
+            }
+        }
+
+        @Override
+        public int getItemCount() {
+            return emails.size() + 2;
+        }
+
+        @Override
+        public int getItemViewType(int position) {
+            if (position == 0 || position == direct_admin_size + 1) return VIEW_TYPE_HEADER;
+            if (position > direct_admin_size + 1) return VIEW_TYPE_USER;
+            else return VIEW_TYPE_ADMIN;
+        }
+    }
+
+    public class AuthEmailViewHolder extends RecyclerView.ViewHolder {
+        View view;
+
+        public AuthEmailViewHolder(View itemView) {
+            super(itemView);
+            view = itemView.findViewById(android.R.id.text1);
+            if (view instanceof Button) {
+                ((Button) view).setOnClickListener(new View.OnClickListener() {
```

Comment:
i'm not sure you need the `Button` cast & check to add a setOnClickListener (I think all views have this method).